src/realm/collection_parent.cpp

```diff
@@ -75,7 +75,7 @@ CollectionParent::~CollectionParent() {}
 
 void CollectionParent::check_level() const
 {
-    if (m_level + 1 > s_max_level) {
+    if (size_t(m_level) + 1 > s_max_level) {
```

Comment:
small suggestions, I saw you are passing the level as uint8_t in many places, probably it is worth to change the type of m_level and s_max_level too...